pkg/gatherers/clusterconfig/dvo_metrics.go

```
@@ -0,0 +1,100 @@
+	"github.com/openshift/insights-operator/pkg/utils/marshal"
+)
+
+const (
+	dvoNamespace = "deployment-validation-operator"
+	// dvo_metrics_endpoint = "http://deployment-validation-operator-metrics.deployment-validation-operator.svc:8383"
+)
+
+var (
+	dvoMetricsServiceNameRegex = regexp.MustCompile(`\bmetrics\b`)
+	dvoMetricsPrefix           = []byte("deployment_validation_operator_")
+)
+
+func (g *Gatherer) GatherDVOMetrics(ctx context.Context) ([]record.Record, []error) {
+	// apiURL, err := url.Parse(dvo_metrics_endpoint)
+	// if err != nil {
+	// 	return nil, []error{err}
+	// }
+	// metricsRESTClient, err := rest.NewRESTClient(apiURL, "/", rest.ClientContentConfig{}, g.gatherKubeConfig.RateLimiter, http.DefaultClient)
+	// if err != nil {
+	// 	klog.Warningf("Unable to load metrics client, no metrics will be collected: %v", err)
+	// 	return nil, nil
+	// }
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherDVOMetrics(ctx, gatherKubeClient.CoreV1(), g.gatherKubeConfig.RateLimiter)
+}
+
+func gatherDVOMetrics(ctx context.Context, coreClient corev1client.CoreV1Interface, rateLimiter flowcontrol.RateLimiter) ([]record.Record, []error) {
+	serviceList, err := coreClient.Services(dvoNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	allDVOMetricsLines := []byte{}
+
+	for _, service := range serviceList.Items {
+		if !dvoMetricsServiceNameRegex.MatchString(service.Name) {
+			continue
+		}
+		for _, port := range service.Spec.Ports {
+			apiURL := url.URL{
+				Scheme: "http",
+				Host:   fmt.Sprintf("%s.%s.svc:%d", service.Name, dvoNamespace, port.Port),
+			}
+			metricsRESTClient, err := rest.NewRESTClient(&apiURL, "/", rest.ClientContentConfig{}, rateLimiter, http.DefaultClient)
+			if err != nil {
+				klog.Warningf("Unable to load metrics client, no metrics will be collected: %v", err)
+				return nil, nil
```

We can return the error here as well.